aws/src/main/java/org/frankframework/filesystem/S3FileRef.java

```diff
@@ -0,0 +1,137 @@
+/*
+   Copyright 2024 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package org.frankframework.filesystem;
+
+import java.io.InputStream;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.frankframework.util.StringUtil;
+
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+
+import jakarta.annotation.Nonnull;
+import lombok.Getter;
+import lombok.Setter;
+
+/**
+ * Object to hold information about an object stored in Amazon S3.
+ * Since the S3 API uses different types of response objects the object here is to:
+ * <ul>
+ * <li>Allow keys to be prefixed with a bucket-name, separated by {@value #BUCKET_OBJECT_SEPARATOR}.</li>
+ * <li>Ensure the correct filename (with extension) is used.</li>
+ * <li>Ensures correct folder names, using UNIX path delimiters, ending with a {@value #FILE_DELIMITER}.</li>
+ * <li>The content-lenght as well as modification-time are stored.</li>
+ * <li>Everything can be updated with a simple method to simplify code in implementations.</li>
+ * </ul>
+ * 
+ * @author Niels Meijer
+ */
+public class S3FileRef {
+
+	private static final String FILE_DELIMITER = "/";
+	public static final String BUCKET_OBJECT_SEPARATOR = "|";
+
+	private final String name;
+	private final String folder;
```

Comment:
Perhaps you can annotate `name` and `folder` with `@Nullable`, so that it is always clear you have to do a null-check with these? Also for code-analysis in the IDE.

Reply:
Name may be empty, but folder can be nullable, I'll add the annotations ;)